File: SSNodeJsChat-Android/app/src/main/java/in/newdevpoint/ssnodejschat/adapter/SelectedContactListAdapter.java
package in.newdevpoint.ssnodejschat.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;



import java.util.ArrayList;

import in.newdevpoint.ssnodejschat.R;
import in.newdevpoint.ssnodejschat.databinding.RowSelectedContactListBinding;
import in.newdevpoint.ssnodejschat.model.ContactModel;

public class SelectedContactListAdapter extends RecyclerView.Adapter<SelectedContactListAdapter.MyHolder> {
	private Context context;
	private ArrayList<ContactModel> tagUserListModelArrayList;
	private OnRemoveIconClickListener onRemoveIconClickListener;

	public SelectedContactListAdapter(Context context, ArrayList<ContactModel> tagUserListModelArrayList, OnRemoveIconClickListener onRemoveIconClickListener) {
		this.context = context;
		this.tagUserListModelArrayList = tagUserListModelArrayList;
		this.onRemoveIconClickListener = onRemoveIconClickListener;
	}

	@NonNull
	@Override
	public MyHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
		RowSelectedContactListBinding binding = DataBindingUtil.inflate(
				LayoutInflater.from(parent.getContext()),
				R.layout.row_selected_contact_list, parent, false);
		return new MyHolder(binding);
	}

	@Override
	public void onBindViewHolder(@NonNull MyHolder holder, final int position) {
		ContactModel tagUserListModel = tagUserListModelArrayList.get(position);
		holder.binding.selectedUserName.setText(tagUserListModel.getFirst_name() + " " + tagUserListModel.getLast_name());
//        if (tagUserListModel.getProfile_image() != null) {
//            if (!(tagUserListModel.getProfile_image().isEmpty())) {
//                Glide.with(context).load(Utils.getImageString(tagUserListModel.getProfile_image())).into(holder.binding.selectedUserImage);
//            } else {
//                holder.binding.selectedUserImage.setImageResource(R.drawable.user_profile_image);
//            }
//        } else
//            holder.binding.selectedUserImage.setImageResource(R.drawable.user_profile_image);

		holder.binding.removeFromTagListbtn.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				onRemoveIconClickListener.OnRemoveUserIconClick(tagUserListModel, position);
			}
		});
	}

	public void addArrayListItem(ContactModel tagUserListModel) {
		this.tagUserListModelArrayList.add(tagUserListModel);
		notifyDataSetChanged();
	}

	@Override
	public int getItemCount() {
		return tagUserListModelArrayList.size();
	}

	public interface OnRemoveIconClickListener {
		void OnRemoveUserIconClick(ContactModel tagUserListModel, int position);
	}

	public class MyHolder extends RecyclerView.ViewHolder {

		private RowSelectedContactListBinding binding;

		public MyHolder(RowSelectedContactListBinding binding) {
			super(binding.getRoot());
			this.binding = binding;
		}

		public void bind(Object obj) {
			binding.executePendingBindings();
		}
	}
}

File: SSNodeJsChat-Android-Kotlin/app/src/main/java/in/newdevpoint/ssnodejschat/adapter/SelectedContactListAdapter.kt
package `in`.newdevpoint.ssnodejschat.adapter

import `in`.newdevpoint.ssnodejschat.R
import `in`.newdevpoint.ssnodejschat.databinding.RowSelectedContactListBinding
import `in`.newdevpoint.ssnodejschat.model.ContactModel
import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import java.util.*

class SelectedContactListAdapter(private val context: Context, tagUserListModelArrayList: ArrayList<ContactModel>, onRemoveIconClickListener: OnRemoveIconClickListener) : RecyclerView.Adapter<SelectedContactListAdapter.MyHolder>() {
    private val tagUserListModelArrayList: ArrayList<ContactModel>
    private val onRemoveIconClickListener: OnRemoveIconClickListener
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyHolder {
        val binding: RowSelectedContactListBinding = DataBindingUtil.inflate(
                LayoutInflater.from(parent.context),
                R.layout.row_selected_contact_list, parent, false)
        return MyHolder(binding)
    }

    override fun onBindViewHolder(holder: MyHolder, position: Int) {
        val tagUserListModel: ContactModel = tagUserListModelArrayList[position]
        holder.binding.selectedUserName.setText(tagUserListModel.first_name + " " + tagUserListModel.last_name)
        //        if (tagUserListModel.getProfile_image() != null) {
//            if (!(tagUserListModel.getProfile_image().isEmpty())) {
//                Glide.with(context).load(Utils.getImageString(tagUserListModel.getProfile_image())).into(holder.binding.selectedUserImage);
//            } else {
//                holder.binding.selectedUserImage.setImageResource(R.drawable.user_profile_image);
//            }
//        } else
//            holder.binding.selectedUserImage.setImageResource(R.drawable.user_profile_image);
        holder.binding.removeFromTagListbtn.setOnClickListener { onRemoveIconClickListener.OnRemoveUserIconClick(tagUserListModel, position) }
    }

    fun addArrayListItem(tagUserListModel: ContactModel) {
        tagUserListModelArrayList.add(tagUserListModel)
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return tagUserListModelArrayList.size
    }

    interface OnRemoveIconClickListener {
        fun OnRemoveUserIconClick(tagUserListModel: ContactModel, position: Int)
    }

    inner class MyHolder(val binding: RowSelectedContactListBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(obj: Any?) {
            binding.executePendingBindings()
        }
    }

    init {
        this.tagUserListModelArrayList = tagUserListModelArrayList
        this.onRemoveIconClickListener = onRemoveIconClickListener
    }
}